tests/PingControllerTest.php

```diff
@@ -0,0 +1,267 @@
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LogLevel;
+use RuntimeException;
+use Symfony\Component\Debug\BufferingLogger;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Throwable;
+use function trigger_error;
+use const E_USER_DEPRECATED;
+use const E_USER_NOTICE;
+
+final class PingControllerTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function it_pings() : void
+    {
+        $controller = new PingController();
+
+        $response = $controller(new Request());
+
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+        $this->assertSame('pong', $response->getContent());
+        $this->assertSame('must-revalidate, no-store, private', $response->headers->get('Cache-Control'));
+        $this->assertSame('text/plain; charset=utf-8', $response->headers->get('Content-Type'));
+        $this->assertFalse($response->headers->has('Expires'));
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_an_expire_header_to_a_http_1_0_request() : void
+    {
+        $controller = new PingController();
+
+        $request = new Request();
+        $request->server->set('SERVER_PROTOCOL', 'HTTP/1.0');
+
+        $response = $controller($request);
+
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+        $this->assertSame('pong', $response->getContent());
+        $this->assertSame('must-revalidate, no-store, private', $response->headers->get('Cache-Control'));
+        $this->assertSame('text/plain; charset=utf-8', $response->headers->get('Content-Type'));
+        $this->assertSame('0', $response->headers->get('Expires'));
+    }
+
+    /**
+     * @test
+     */
+    public function it_pings_with_a_check() : void
+    {
+        $logger = new BufferingLogger();
+        $count = 0;
+        $check = function () use (&$count) : void {
+            $count++;
+        };
+
+        $controller = new PingController($check, $logger);
+
+        $response = $controller(new Request());
+
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
+        $this->assertSame('pong', $response->getContent());
+        $this->assertSame('must-revalidate, no-store, private', $response->headers->get('Cache-Control'));
+        $this->assertSame('text/plain; charset=utf-8', $response->headers->get('Content-Type'));
+        $this->assertCount(0, $logger->cleanLogs());
+        $this->assertSame(1, $count);
+    }
+
+    /**
+     * @test
+     */
+    public function it_fails_when_there_is_a_runtime_exception() : void
+    {
+        $exception = new RuntimeException('Problem');
+
+        $controller = new PingController($this->willThrow($exception));
+
+        $response = $controller(new Request());
+
+        $this->assertSame(Response::HTTP_SERVICE_UNAVAILABLE, $response->getStatusCode());
+        $this->assertSame('Service Unavailable', $response->getContent());
```

Comment:
Should these look like normal API errors?

Reply:
Given the positive response is just `text/plain`, it fits that these are text too. Sensitive information is in exception message and stack traces, hence it is usually not provided in the response; once a bad status code is known a client (Docker healthcheck, load balancer, CDN) won't probably do any parsing of the body.